src/maheshKJun20/ValidateAtmPin.java

```diff
@@ -0,0 +1,79 @@
+package maheshKJun20;
+
+public class ValidateAtmPin {
+	
+	void validatePin(String pin) {
+		if (validateLength(pin)) {
+			if (validateDigit(pin)) {
+				if (validateNumRepitition(pin))
+					if (validateAscendingOrder(pin) || validateDescendingOrder(pin))
+						System.out.println("ATM pin " + pin + " is Valid");
+					else
+						System.out.println("Invalid pin. Order Should be Ascending or Desending");
+			}				
+		}		
+	}
+	
+	boolean validateLength(String pin) {
+		if (pin.length() == 4)
+			return true;
+		else {
+			System.out.println("Invalid pin. Length of pin: " + pin.length());
+			return false;
+		}
+	}
+	
+	boolean validateDigit(String pin) {
+		for (int index=0; index<pin.length();index++) {
+			if(!Character.isDigit(pin.charAt(index))){
+				System.out.println("Invalid pin. Invalid character in the pin '" + pin.charAt(index) + "'");
+				return false;
+			}
+		}
+		return true;
+	}
+	
+	boolean validateAscendingOrder(String pin) {
+		for (int index=0; index<pin.length();index++) {
+			int minAscii = (int)pin.charAt(index);
+			for (int innerIndex=index+1; innerIndex<pin.length();innerIndex++) {
+				if (!(minAscii < (int)pin.charAt(innerIndex))) {
+					return false;
+				}
+			}
+		}
+		return true;
+	}
+	
+	boolean validateDescendingOrder(String pin) {
+		for (int index=0; index<pin.length();index++) {
+			int maxAscii = (int)pin.charAt(index);
+			for (int innerIndex=index+1; innerIndex<pin.length();innerIndex++) {
+				if (!(maxAscii > (int)pin.charAt(innerIndex))) {
+					return false;
+				}
+			}
+		}
+		return true;
+	}
+	
+	boolean validateNumRepitition(String pin) {
+		for (int index=0; index<pin.length();index++) {
+			for (int innerIndex=index+1; innerIndex<pin.length();innerIndex++) {
```

Comment:
why do we need two for loop here?
we can do using one right

use indexOf & lastIndexOf

Reply:
@harshvegada  - Yes, one loop with indexOf and lastIndexOf method would have been a better solution to this. Noted and will amend the changes. Thanks for the feeback